c10/util/numa.cpp

```diff
@@ -0,0 +1,137 @@
+#include "c10/util/numa.h"
+
+#if defined(__linux__) && !defined(C10_DISABLE_NUMA) && C10_MOBILE == 0
+#include <numa.h>
+#include <numaif.h>
+#include <unistd.h>
+#define C10_NUMA_ENABLED
+#endif
+
+namespace c10 {
+
+#ifdef C10_NUMA_ENABLED
+bool IsNUMAEnabled() {
+  return numa_available() >= 0;
+}
+
+void NUMABind(int numa_node_id) {
+  if (numa_node_id < 0) {
+    return;
+  }
+  if (!IsNUMAEnabled()) {
+    VLOG(1) << "NUMA is not enabled";
+    return;
+  }
+
+  AT_CHECK(
+      numa_node_id <= numa_max_node(),
+      "NUMA node id ", numa_node_id, " is unavailable");
+
+  auto bm = numa_allocate_nodemask();
+  numa_bitmask_setbit(bm, numa_node_id);
+  numa_bind(bm);
+  numa_bitmask_free(bm);
+}
+
+int GetNUMANode(const void* ptr) {
+  if (!IsNUMAEnabled()) {
+    VLOG(1) << "NUMA is not enabled";
+    return {};
```

Comment:
Err... I don't think you wanted to return {}, if your return type is an int.

Reply:
we have a followup diff that changes the type to optional<int> and we're returning c10::nullopt there.

Reply:
Yes, but if you want to stack your diffs, each intermediate diff has to be correct...

Reply:
this means 0 for int? is it?

Reply:
Yes, and before it was -1

Reply:
@ezyang actually I just copied the code, so it's 0 before as well.  what should we return here?
cc @ilia-cher